[CI] VerifyVersionConstantsIT testLuceneVersionConstant failing

**Build Scans:**
- [elasticsearch-intake #24360 / 8.19.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/hgbcyrqurwq7o)
- [elasticsearch-pull-request #76815 / 8.19.0_part_6_bwc-snapshots](https://gradle-enterprise.elastic.co/s/v4svffbva7thc)
- [elasticsearch-pull-request #76813 / 8.19.0_bwc](https://gradle-enterprise.elastic.co/s/e4zfffs6bhmt6)

**Reproduction Line:**
```
./gradlew ":qa:verify-version-constants:v8.19.0#bwcTest" -Dtests.class="org.elasticsearch.qa.verify_version_constants.VerifyVersionConstantsIT" -Dtests.method="testLuceneVersionConstant" -Dtests.seed=442A2DB270FC8655 -Dtests.bwc=true -Dtests.locale=cy -Dtests.timezone=GB-Eire -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.qa.verify_version_constants.VerifyVersionConstantsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testLuceneVersionConstant),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <9.12.2>
     but: was <9.12.1>
```

**Issue Reasons:**
- [main] 3 failures in test testLuceneVersionConstant (2.0% fail rate in 151 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -150,9 +150,6 @@ tests:
 - class: org.elasticsearch.xpack.security.SecurityRolesMultiProjectIT
   method: testUpdatingFileBasedRoleAffectsAllProjects
   issue: https://github.com/elastic/elasticsearch/issues/129775
-- class: org.elasticsearch.qa.verify_version_constants.VerifyVersionConstantsIT
-  method: testLuceneVersionConstant
-  issue: https://github.com/elastic/elasticsearch/issues/125638
 - class: org.elasticsearch.gradle.internal.InternalDistributionBwcSetupPluginFuncTest
   method: "builds distribution from branches via archives extractedAssemble [bwcDistVersion: 8.2.1, bwcProject: bugfix, expectedAssembleTaskName:
     extractedAssemble, #2]"

